### PR TITLE
Extrude/Interaction Fixes

### PR DESCRIPTION
**Pull request by DTilley22**
- Increased default radius size
- Decreased the rate which the radius changes
- Lowered material opacity for objects that are in radius but not the closest
- Removed mesh/mesh filter from face handles
- Hovering over a face will highlight its edges and vertices, moving it will highlight the edges
- Removed removeVisuals() call from extrude
- Added createVisuals() in extrude that works the same as in MeshRebuilder but immediately destroys duplicates of existing gameobjects
- Extrusion no longer unlocks vertices and can be done on faces containing locked vertices

- Using extrusion with the secondary button will move the new face with the hand, press it once to extrude and move and press it again to stop moving

Bugs/Issues
- This was from the ChangeMatClosest rework but when the radius is much smaller than an edge it won't highlight correctly
- CreateVisuals() in extrude isn't optimized, it's easy to instantiate new vertices and faces during extrusion but I couldn't figure out how to instantiate new edges easily without using the code from MeshRebuilder
- Tried to get the secondary extrusion to stop with the right trigger as well but it wasn't being detected, it shouldn't be too hard though

### CODE REVIEW

File: Assets/Scripts/Tools/Extrude.cs [hunk 1]
```diff
@@ -222,6 +243,164 @@ public void OnTriggerEnter(Collider other)
...
+            // Connect a line from our starting vertex to each adjacent vertex
+            foreach (int k in adjacentVertices)
+            {
+                // Ignore adjacent vertices we've already dealt with
+                if (k < i)
+                    continue;
+
+                // Same as vertex, create a new edge object and set its parent
+                GameObject newEdge = Instantiate(edge, meshRebuilder.model.transform);
+                Component[] components = newEdge.GetComponents(typeof(Component));
```
Comment (Rami-J) [inline Assets/Scripts/Tools/Extrude.cs:306]: Is this needed? If not, we can probably remove the logging and GetComponents() method

**Review by Rami-J — COMMENTED** [review]
Bugs look like they were fixed. Only glaring issue is the performance on extrude. It seems like the performance somehow got worse? I tested it on a ~60kb file and it freezes my game everytime I try extruding

**Comment by Rami-J** [comment]
The .Remove() calls in the meshRebuilder might be hurting the runtime since it's a linear operation on arrays. Would probably be faster if everything was in a dictionary of index -> Vertex/Edge/Face object instead but idk if this will mess with other existing things.

File: Assets/Scripts/Tools/Extrude.cs (hunk 1 above)
Comment (Rami-J) [reply Assets/Scripts/Tools/Extrude.cs:306]: I tested removing this and I think this was the culprit for the worse performance actually. I'll push a commit to remove it if we don't need it.

**Review by Rami-J — APPROVED** [review]

**Comment by DTilley22** [comment]
> ~Bugs look like they were fixed. Only glaring issue is the performance on extrude. It seems like the performance somehow got worse? I tested it on a ~60kb file and it freezes my game for a noticeable time everytime I try extruding~ The performance returned back to how it was I think after removing an unecessary GetComponents() call

Yeah sorry that was for debugging I forgot to remove that line

**Comment by Rami-J** [comment]
I think the most recent commit broke extrusion on multiplayer

**Comment by Rami-J** [comment]
transform.localPosition assign attempt for 'Edge(Clone)' is not valid. Input localPosition is { Infinity, -Infinity, Infinity }.
UnityEngine.Transform:set_localPosition (UnityEngine.Vector3)
Extrude:CreateVisuals (MeshRebuilder,System.Collections.Generic.List`1<UnityEngine.Vector3>,System.Collections.Generic.List`1<int>,int,int) (at Assets/Scripts/Tools/Extrude.cs:341)
Extrude:extrudeFace (int,MeshRebuilder,UnityEngine.Mesh,bool) (at Assets/Scripts/Tools/Extrude.cs:220)
EasyMeshVR.Multiplayer.NetworkMeshManager:HandleMeshFaceExtrudeEvent (EasyMeshVR.Multiplayer.FaceExtrudeEvent) (at Assets/Scripts/Networking/NetworkMeshManager.cs:542)
EasyMeshVR.Multiplayer.NetworkMeshManager:HandleMeshFaceExtrudeEvent (object[]) (at Assets/Scripts/Networking/NetworkMeshManager.cs:525)
EasyMeshVR.Multiplayer.NetworkMeshManager:OnEvent (ExitGames.Client.Photon.EventData) (at Assets/Scripts/Networking/NetworkMeshManager.cs:328)

File: Assets/Scripts/Tools/Extrude.cs [hunk 2]
```diff
@@ -70,37 +58,59 @@ public override void PrimaryAction()
...
-        //     return;
-        // }
-
         // calculating normals is done in mesh rebuilder
 
         // align new vertices w/ face normal
-        Vector3 new1 = vertex1.transform.localPosition + ((faceObj.normal  + vertex1.transform.localPosition *.005f).normalized );
-        Vector3 new2 = vertex2.transform.localPosition + ((faceObj.normal  + vertex2.transform.localPosition *.005f).normalized );
-        Vector3 new3 = vertex3.transform.localPosition + ((faceObj.normal  + vertex3.transform.localPosition *.005f).normalized );
+        Vector3 new1 = vertex1.transform.localPosition + ((faceObj.normal  + vertex1.transform.localPosition *.005f).normalized ) / distance;
```
Comment (Rami-J) [inline Assets/Scripts/Tools/Extrude.cs:107]: i think we are dividing by zero here in the case where the extrude event is sent to the other players

Comment (Rami-J) [reply Assets/Scripts/Tools/Extrude.cs:107]: one way to fix this error is to send the distance in the event

**Review by Rami-J — APPROVED** [review]
Fixed the crashing in multiplayer

**Pull request merged**